components/net/http_loader.rs

```diff
@@ -504,7 +504,8 @@ fn request_must_be_secured(url: &Url, hsts_list: &Arc<RwLock<HSTSList>>) -> bool
 pub fn modify_request_headers(headers: &mut Headers,
                               doc_url: &Url,
                               user_agent: &str,
-                              cookie_jar: &Arc<RwLock<CookieStorage>>) {
+                              cookie_jar: &Arc<RwLock<CookieStorage>>,
+                              load_data: LoadData) {
```

Comment:
This should probably take an `&LoadData` so we don't have to clone it. Alternatively, just pass in `include_third_party_credentials` directly